### PR TITLE
[WRAPPER] Used RunFunctionFmt for better sign-extend handling

### PR DESCRIPTION
**Pull request by ksco**

### CODE REVIEW

**Comment by ksco** [comment]
TODOs:
1. Double check. It's all by hand and it's huge, so high probability that there are some typos.

Future TODOs:
1. Wrapperhelper/gen.cpp should generate `RunFunctionFmt`;
3. `wrappedudev0.c` and `wrappedudev1.c` are left unconverted, don't know how to handle VARARGS.

File: src/libtools/sdl2rwops.c [hunk 1]
```diff
@@ -85,36 +85,36 @@ EXPORT int32_t my2_native_close(SDL2_RWops_t *context)
 }
 EXPORT int64_t my2_emulated_size(SDL2_RWops_t *context)
 {
-    return (int64_t)RunFunction(my_context, (uintptr_t)context->hidden.my.orig->size, 1, context->hidden.my.orig);
+    return (int64_t)RunFunctionFmt(my_context, (uintptr_t)context->hidden.my.orig->size, "p", context->hidden.my.orig);
 }
 EXPORT int64_t my2_emulated_seek(SDL2_RWops_t *context, int64_t offset, int32_t whence)
 {
-    return (int64_t)RunFunction(my_context, (uintptr_t)context->hidden.my.orig->seek, 3, context->hidden.my.orig, offset, whence);
+    return (int64_t)RunFunctionFmt(my_context, (uintptr_t)context->hidden.my.orig->seek, "pLi", context->hidden.my.orig, offset, whence);
```
Comment (ptitSeb) [inline src/libtools/sdl2rwops.c:92]: should be "pli" because it's signed long

Comment (ptitSeb) [reply src/libtools/sdl2rwops.c:92]: also, probably more like "pIi" because it's an explicit 64bits value iirc

Comment (ksco) [reply src/libtools/sdl2rwops.c:92]: There must be a lot of errors like this, you might want to review after I double-checked.

Comment (ksco) [reply src/libtools/sdl2rwops.c:92]: I'm afk for an hour or so, I'll do the check later.

**Review by ksco — COMMENTED** [review]
Ok, I've reviewed, I'm confident now.

File: src/libtools/sdl2rwops.c [hunk 2]
```diff
@@ -85,36 +85,36 @@ EXPORT int32_t my2_native_close(SDL2_RWops_t *context)
 }
 EXPORT int64_t my2_emulated_size(SDL2_RWops_t *context)
 {
-    return (int64_t)RunFunction(my_context, (uintptr_t)context->hidden.my.orig->size, 1, context->hidden.my.orig);
+    return (int64_t)RunFunctionFmt(my_context, (uintptr_t)context->hidden.my.orig->size, "p", context->hidden.my.orig);
 }
 EXPORT int64_t my2_emulated_seek(SDL2_RWops_t *context, int64_t offset, int32_t whence)
 {
-    return (int64_t)RunFunction(my_context, (uintptr_t)context->hidden.my.orig->seek, 3, context->hidden.my.orig, offset, whence);
+    return (int64_t)RunFunctionFmt(my_context, (uintptr_t)context->hidden.my.orig->seek, "pli", context->hidden.my.orig, offset, whence);
```
Comment (ptitSeb) [inline src/libtools/sdl2rwops.c:92]: I still prefer 'I' (capital i) for int64_t, more accurate than 'l' (small L)

File: src/tools/gtkclass.c [hunk 3]
```diff
@@ -2059,12 +2059,12 @@ static void bridgeGtkGrid3Class(my_GtkGrid3Class_t* class)
...
 // wrapper x86 -> natives of callbacks
-WRAPPER(GtkEventController, set_widget, void, (void* controller, void* widget), 2, controller, widget);
-WRAPPER(GtkEventController, unset_widget, void, (void* controller), 1, controller);
-WRAPPER(GtkEventController, handle_event, int, (void *controller, void *event, double x, double y), 4, controller, event, x, y);
-WRAPPER(GtkEventController, reset, void, (void* controller), 1, controller);
-WRAPPER(GtkEventController, handle_crossing, void, (void *controller, void *crossing, double x, double y), 4, controller, crossing, x, y);
-WRAPPER(GtkEventController, filter_event, void, (void *controller, void *event), 2, controller, event);
+WRAPPER(GtkEventController, set_widget, void, (void* controller, void* widget), "pp", controller, widget);
+WRAPPER(GtkEventController, unset_widget, void, (void* controller), "p", controller);
+WRAPPER(GtkEventController, handle_event, int, (void *controller, void *event, double x, double y), "ppdd", controller, event, x, y);
```
Comment (ptitSeb) [inline src/tools/gtkclass.c:2064]: All thoses functions with `double` parameters were probably not working fine before also! That will improve Gtk support :)

File: src/libtools/sdl2rwops.c (hunk 2 above)
Comment (ksco) [reply src/libtools/sdl2rwops.c:92]: Ah, sorry, I thought it’s a small L.. They looks similar!

File: src/tools/gtkclass.c (hunk 3 above)
Comment (ksco) [reply src/tools/gtkclass.c:2064]: Yes, they're expected to be passed through the float registers!

**Review by ptitSeb — COMMENTED** [review]
the "U" / "L" are not important in 64bits, but are in 32bits. I would like this work to be reusable in 32bits if needed.

**Comment by ksco** [comment]
Thanks for the detailed review! I’ll not be available for the next few hours, you can merge this and change it yourself if you want..

**Comment by ksco** [comment]
I've replaced all `U` with `L` for `unsigned long`, and also used `l` instead of `L` for `off_t`.

**Pull request merged**